pkg/kn/commands/flags/sink.go

```diff
@@ -84,7 +88,10 @@ func (i *SinkFlags) ResolveSink(knclient clientdynamic.KnDynamicClient, namespac
 	}
 	typ, ok := sinkMappings[prefix]
 	if !ok {
-		return nil, fmt.Errorf("unsupported sink type: %s", i.sink)
+		if prefix == "svc" || prefix == "service" {
```

Comment:
I have not tested it but does this also throw an error if a user configures that prefix in its configuration file ? (that should not be the case).

Reply:
Nope, this check is after key lookup in `sinkMapping` (`	typ, ok := sinkMappings[prefix]`) just a couple lines ago.